src/envoy/http/authn/authenticator_base.cc

```diff
@@ -68,7 +73,21 @@ bool AuthenticatorBase::validateX509(const iaapi::MutualTls& mtls,
 bool AuthenticatorBase::validateJwt(const iaapi::Jwt& jwt, Payload* payload) {
   std::string jwt_payload;
   if (filter_context()->getJwtPayload(jwt.issuer(), &jwt_payload)) {
-    return AuthnUtils::ProcessJwtPayload(jwt_payload, payload->mutable_jwt());
+    std::string payload_to_process = jwt_payload;
+    std::string original_payload;
+    bool found = false;
+    for (auto h : jwt.jwt_headers()) {
+      if (kExchangedTokenHeaderName == h) {
+        found = true;
+        break;
+      }
+    }
+    if (found &&
+        AuthnUtils::ExtractOriginalPayload(jwt_payload, &original_payload)) {
```

Comment:
If it is APToken (found), but cannot extract original payload, then probably should return false.

Reply:
Limin mentioned that an APToken may not necessarily contain the original payload (i.e., original_claims). If an APToken must contain the original payload, we can use it as a way to tell whether a token is an APToken. Hence, no false is returned here.

Reply:
Is that so? Again, I feel wrong to while making assumption of the location, but not on the token format and it's behavior. Anyway, please add comment to explain the rationale.

Reply:
Done: added comments to explain the rationale.

Reply:
I think for the first version, it is ok to assume that AP token always carries original_claims. This may not be true going forward, but let's focus on this JWT token exchange use case for now.